### DISCUSSION TITLE
Request for typedefs for "custom stream" callbacks

### DISCUSSION THREAD

**Discussion by lopcode** [libvips/libvips#4186] [Ideas]
Hi there 👋 I just finished adding "custom streams" support to [vips-ffm](https://github.com/lopcode/vips-ffm), to allow for streaming from/to Java InputStream/OutputStream types. One slightly annoying thing I had to do is add my own C typedefs, to allow jextract (an automated binding generator) to generate otherwise fiddly callback marhsalling code: https://github.com/lopcode/vips-ffm/blob/main/jextract_entry/callback_typedefs.h

Would it be possible (and make sense) to add these typedefs directly in the libvips includes? (Or have I missed them somewhere?)

**Comment by jcupitt** [comment]
Hi again,

That's great! 

I don't think we'd use the typedefs in libvips, but I agree they'd be handy for bindings. I think pyvips could also use them, for example. 

Do whip up a small PR if you have time.
